apps/src/lib/node/ledger/shell/queries.rs

```diff
@@ -321,6 +321,23 @@ pub(crate) trait QueriesExt {
         tm_address: &[u8],
         epoch: Option<Epoch>,
     ) -> std::result::Result<Address, Error>;
+
+    /// Determines if it is possible to send a validator set update vote
+    /// extension at the provided [`BlockHeight`].
+    ///
+    /// This is done by checking if we are at the first block of a new epoch,
+    /// or if we are at block height 1 of the first epoch.
```

Comment:
no, since the first block will have height 0 (genesis), we will not have vote extensions. so, we can only send a validator set update vote extension starting from height 1. basically, epoch 0 is an edge case. for all other epochs, we can send validator set update vote extensions at the first block of the epoch.

Reply:
The first ever block (genesis block) should have height 1. There doesn't exist a block with height 0, but I think I see what you mean. 

So when proposing block #0 of epoch #0 (i.e. the genesis block, absolute height 1), we don't have vote extensions as there was no previous Tendermint round. So the block proposer should create the valset update during the proposal for block #0 in that case. 

In later epochs, validators should send their valset update during the proposal for block #0, then the block proposer would aggregate them in the proposal for block #1 i.e. the second block proposed for an epoch? Based on the vote extensions from the proposal for the first block of the epoch.

In practice we will not be starting the bridge at genesis (i.e. during epoch #0) but I guess the same logic may apply around whatever epoch we start the bridge. I'd update this docstring to mention proposing e.g. "if we are proposing the second block of an epoch, or if we are proposing the first block of the first epoch"(?)